pytorch_lightning/utilities/auto_restart.py

```diff
@@ -476,6 +430,48 @@ def _capture_metadata_collate(
     return {"data": data, AutoRestartBatchKeys.PL_RESTART_META: metadata}
 
 
+# TODO: Merge this code within stateful DataLoaderIter.
```

Comment:
curious: is this something you plan to do shortly or a "would be nice"

Reply:
Shortly.